### PR TITLE
[5G: MVC][SessionD][Pipelined] Re-Definition of Pipelined and Session manager protos

### PR DESCRIPTION
**Pull request by ronit-kumar-acl**
Signed-off-by: ronit-kumar-acl <ronit.k@altencalsoftlabs.com>

## Summary

Contains latest protos which have been concluded recently - 
Major Changes: 
1. NodeId moved to session_manager
2. QER support in pipelined using existing ActivateFlowsRequest
3. Flattened SessionSet message
4. Pdr State added
5. Renaming of few fields like session_version.
## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/protos/pipelined.proto [hunk 1]
```diff
@@ -200,36 +200,19 @@ message PacketDropTableId {
...
-        SESSION_ACTIVE                        = 0;
-        SESSION_TERMINATED                    = 4;
-        SESSION_RELEASED                      = 6;
-        CREATING                              = 7;
-        CREATED                               = 8;
-        ACTIVE                                = 9;
-        INACTIVE                              = 10;
-        RELEASE                               = 11;
+    UNUSED       = 0;
+    //Adhering to current refactored converged sessiond states
```
Comment (themarwhal) [inline lte/protos/pipelined.proto:209]: 👍

File: lte/protos/pipelined.proto [hunk 2]
```diff
@@ -200,36 +200,19 @@ message PacketDropTableId {
...
+    CREATING     = 7;
+    CREATED      = 8;
+    ACTIVE       = 9;
+    INACTIVE     = 10;
+    RELEASE      = 11;
```
Comment (uri200) [inline lte/protos/pipelined.proto:214]: is it possible to reset the identifiers and start from index 1? 
Per my understanding `FsmState` is not used by any of our systems on production yet, so no backwards compatibility

File: lte/protos/pipelined.proto [hunk 3]
```diff
@@ -246,57 +229,31 @@ message SdfFilters {                               //8.2.5
...
+  NOCP = 3;
+  DUPL = 4;
 }
 
 message RedirectInfo {                //8.2.20
   enum RedirectAddrType {
-    IPV4 = 0;
-    IPV6 = 1;
+    IPV4_0 = 0;
+    IPV6_0 = 1;
```
Comment (uri200) [inline lte/protos/pipelined.proto]: is this right?
Isn't this proto being used currently? If so, we shouldn't change the name

File: lte/protos/pipelined.proto (hunk 2 above)
Comment (themarwhal) [reply lte/protos/pipelined.proto:214]: +1 on this.

**Review by themarwhal — APPROVED** [review]
Please address some of the comments by @uri200 . The other changes look ok to me. CC @pshelar for PipelineD perspective

**Comment by themarwhal** [comment]
@ronit-kumar-acl Please commit the generated files as well.

File: lte/protos/pipelined.proto (hunk 2 above)
Comment (ronit-kumar-acl) [reply lte/protos/pipelined.proto:214]: @gurrapualt Can you please elaborate reason for using these enums with the above indexing ?

File: lte/protos/pipelined.proto (hunk 3 above)
Comment (ronit-kumar-acl) [reply lte/protos/pipelined.proto]: RedirectInfo has been added by SMF team, so we are in sync with UPF team regarding the same. I suppose it's not being used by the legacy code too so shouldn't be any issue.

**Comment by ronit-kumar-acl** [comment]
> @ronit-kumar-acl Please commit the generated files as well.

Hi @themarwhal , sure I'll do the same.

File: lte/protos/session_manager.proto [hunk 4]
```diff
@@ -783,6 +794,23 @@ message SetSMSessionContextAccess {
   RatSpecificContextAccess   rat_specific_context = 2;
 }
 
+
+message UPFSessionState {
+  string seid = 1;            // Session ID
```
Comment (ulaskozat) [inline lte/protos/session_manager.proto]: is this really the session id? or is it the session key?

File: lte/protos/pipelined.proto (hunk 3 above)
Comment (ulaskozat) [reply lte/protos/pipelined.proto]: why do we need to change the enum names here?

File: lte/protos/session_manager.proto (hunk 4 above)
Comment (ulaskozat) [reply lte/protos/session_manager.proto]: we need to be very explicit here. e.g., if we are passing the tunnel ID or IMSI, etc., let's explicitly define the field for that.

**Review by ulaskozat — CHANGES_REQUESTED** [review]
Few comments are unresolved.

File: lte/protos/session_manager.proto (hunk 4 above)
Comment (ronit-kumar-acl) [reply lte/protos/session_manager.proto]: @gurrapualt @panyogesh can we change this keyword to what we have finalized ?

File: lte/protos/pipelined.proto (hunk 3 above)
Comment (ronit-kumar-acl) [reply lte/protos/pipelined.proto]: I checked, earlier the enums  with name IPV4 and IPV6 were conflicting with an already defined enum. But just now I checked and that issue has been resolved hence I'll remove _0 part from the enums.

File: lte/protos/session_manager.proto [hunk 5]
```diff
@@ -785,6 +796,23 @@ message SetSMSessionContextAccess {
   RatSpecificContextAccess   rat_specific_context = 2;
 }
 
+
+message UPFSessionState {
+  string imsi = 1;            
+  int32 sess_ver_no = 2;      // Successful Version number snapshot from UPF. If its not the expected parse Failure report.
+}
```
Comment (ulaskozat) [inline lte/protos/session_manager.proto:803]: isn't the agreement was to send the local tunnel id to identify the session?

Comment (panyogesh) [reply lte/protos/session_manager.proto:803]: Ulas. That's right (reference : #2971). Will sync up with SMF team if they can update this.

**Comment by gurrapualt** [comment]
This code pulled as part of #3366 request, so closing this.

**Pull request closed without merging**